First version of example.py

matkrak opened the pull request:
The domain was created with range() because from this url it looks like:
YEAR MONTH
 1750       1
 1750       2
     .          . 
     .          .

matkrak committed:
First version of example.py

The domain was created with range() because from this url it looks like:
YEAR MONTH
 1750       1
 1750       2
     .          . 
     .          .

grzanka commented on beprof/example.py:
```diff
@@ -0,0 +1,20 @@
+
+data = urllib.request.urlopen(url)
+
+#skip lines with plain text
```
RTFM (skip_header)
http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.genfromtxt.html

matkrak committed:
Fixed

I tried to use skip_header and skip_footer yesterday but it didn't work so I abbandoned it. I don't know what I was doing wrong but it works OK now.

grzanka commented on beprof/example.py:
```diff
@@ -0,0 +1,15 @@
+
+all = np.genfromtxt(data, skip_header=35, skip_footer=2192)
+print(np.shape(all))
+x = np.array(range(0, 1000))
```
Again RTFM.
http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.arange.html

Before coding any new line in Python, go to numpy library reference, check which functions are available there and what are they doing. Print such list above your desk. It might help you.

grzanka commented on beprof/example.py:
```diff
@@ -0,0 +1,15 @@
+x = np.array(range(0, 1000))
+y = all[:,3]
+
+example = Curve(np.stack((x, y), axis=1))
```
Whole idea behind this library is to deal with data which have X and Y component.
If you have only 1-dimensional data, you can do all such tasks with numpy itself, without much effort.

You should find such dataset which contains X and Y data. X shouldn't be simply a counter running from zero.

grzanka replied:
Issue #9 is about Profile example, not Curve example

matkrak replied:
I thought it was more about using urllib and parsing the data but yes, it says Profile example so I will find proper data and change Curve into Profile.

grzanka commented on beprof/example.py:
```diff
@@ -0,0 +1,15 @@
+from beprof.curve import Curve
+import numpy as np
+
+url = 'http://berkeleyearth.lbl.gov/auto/Global/Complete_TAVG_complete.txt'
```
Does this dataset contain data sample which is rising and falling ? In the comment to #9 I said "see for example", not "use it".

matkrak replied:
OK I will find better example. By rising and falling you mean it should be unimodal, or it can have many extremums but it should "start" with a rising edge?

grzanka replied:
A good example could be noisy parabola:

`0.01*f(x) + 1 - x^2`

where f(x) gives random real numbers from interval [-1,1]

matkrak committed:
Sorry

I must agree, that reading manual can be helpful.

matkrak committed:
Blank lines removed

grzanka commented:
I just wanted to recall fragment of discussion on the issue:
https://github.com/grzanka/beprof/issues/9

> We can start working on it inside example.py file in separate branch. 
> At the beginning of development it will be long and dirty.
> When its mature enough I'd like to put the code snippet in README.md file and remove example.py file.
> 
> What user installs ad pip package shouldn't contain any example.
> You can keep them in repo, but them you should filter the data to package in setup.py

grzanka commented:
Closing, as promised. In case of resuming activity, open new PR.
